src/Orchard.Web/Modules/Orchard.Email/Services/SmtpMessageChannel.cs

```diff
@@ -133,35 +124,68 @@ public class SmtpMessageChannel : Component, ISmtpChannel, IDisposable {
         }
 
         private SmtpClient CreateSmtpClient() {
+            var smtpConfiguration = new {
+                _smtpSettings.Host,
+                _smtpSettings.Port,
+                _smtpSettings.EncryptionMethod,
+                _smtpSettings.AutoSelectEncryption,
+                _smtpSettings.RequireCredentials,
+                _smtpSettings.UserName,
+                _smtpSettings.Password,
+            };
             // If no properties are set in the dashboard, use the web.config value.
             if (String.IsNullOrWhiteSpace(_smtpSettings.Host)) {
-                return new SmtpClient(); 
-            }
+                var smtpSection = (SmtpSection)ConfigurationManager.GetSection("system.net/mailSettings/smtp");
+                if (smtpSection.DeliveryMethod != SmtpDeliveryMethod.Network) {
+                    throw new NotSupportedException($"Only the {SmtpDeliveryMethod.Network} delivery method is supported, but "
+                        + $"{smtpSection.DeliveryMethod} delivery method is configured. Please check your Web.config.");
+                }
 
-            var smtpClient = new SmtpClient {
-                UseDefaultCredentials = _smtpSettings.RequireCredentials && _smtpSettings.UseDefaultCredentials
-            };
+                smtpConfiguration = new {
+                    smtpSection.Network.Host,
+                    smtpSection.Network.Port,
+                    EncryptionMethod = smtpSection.Network.EnableSsl ? SmtpEncryptionMethod.SslTls : SmtpEncryptionMethod.None,
+                    AutoSelectEncryption = !smtpSection.Network.EnableSsl,
+                    RequireCredentials = smtpSection.Network.DefaultCredentials || !String.IsNullOrWhiteSpace(smtpSection.Network.UserName),
+                    smtpSection.Network.UserName,
+                    smtpSection.Network.Password,
+                };
+            }
 
-            if (!smtpClient.UseDefaultCredentials && !String.IsNullOrWhiteSpace(_smtpSettings.UserName)) {
-                smtpClient.Credentials = new NetworkCredential(_smtpSettings.UserName, _smtpSettings.Password);
+            var secureSocketOptions = SecureSocketOptions.Auto;
```

Comment:
SecureSocketOptions defaulting to None may be a compatibility issue when SmtpSettingsPart.EnableSsl was previously set to true (the value of which is now based on SmtpSettingsPart.EncryptionMethod value only). In those cases, Ssl would be disabled. Can a migration properly set the EncryptionMethod property value to e.g. SmtpEncryptionMethod.SslOnConnect when "old" EnableSsl was true?

Reply:
@dministro how did we handle this for our sites? Anyway, a migration step for keeping the correct settings would be nice indeed.

Reply:
This is handled a bit differently on our sites. I kept the `EnableSsl` and the [MailKit.MAilService.Connect(string host, int port, bool useSsl, CancellationToken cancellationToken)](https://github.com/jstedfast/MailKit/blob/7f3affdb9716bf84270f7d831e90a42a89c9d7f8/MailKit/MailService.cs#L975-L1035) is used to initiate the connection.

The solution implemented under this PR gives a bit more control over the connection settings.

Reply:
Yeah, I was wondering, because the UI is different compared to our sites in Production now, but it's good to have better configurability. At the same time, what Andrea is saying is that the configuration that was set previously (including SSL and authentication) should carry over.

Reply:
I added the migration step to migrate from `EnableSsl` to `EncryptionMethod`. See https://github.com/OrchardCMS/Orchard/pull/8707/commits/8a6958cec2edf28bfdabaadd19088c7a866be963

Regarding the `UseDefaultCredentials` there is no alternative in `MailKit` so removing this is a breaking change. Here is how this situation is handled in `OrchardCore.Email` module:

https://github.com/OrchardCMS/OrchardCore/blob/8cc817c13d12ec39f4dbdec6f9c71195c0c511c1/src/OrchardCore/OrchardCore.Email.Core/Services/SmtpService.cs#L295-L299

Reply:
I pushed some changes to simplify how the settings are migrated + the settings upgrade also includes setting AutoSelectEncryption to the opposite of what EnableSsl was (so it's more explicit).